src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp

```diff
@@ -347,53 +350,66 @@ namespace OpenMS
     }
     n_external_peps_ = peptide_map_.size() - n_internal_peps_;
 
+
     OPENMS_LOG_INFO << "Creating assay library..." << endl;
-    PeptideRefRTMap ref_rt_map;
-    createAssayLibrary_(peptide_map_, ref_rt_map);
+    boost::shared_ptr<PeakMap> shared = boost::make_shared<PeakMap>(ms_data_);
+    OpenSwath::SpectrumAccessPtr spec_temp =
+        SimpleOpenMSSpectraFactory::getSpectrumAccessOpenMSPtr(shared);
+    auto chunks = chunk_(peptide_map_.begin(), peptide_map_.end(), batch_size_);
 
-    if (debug_level_ >= 666)
-    {
-      cout << "Writing debug.traml file." << endl; 
-      TraMLFile().store("debug.traml", library_);
-    }
 
-    //-------------------------------------------------------------
-    // run feature detection
-    //-------------------------------------------------------------
-    OPENMS_LOG_INFO << "Extracting chromatograms..." << endl;
-    ChromatogramExtractor extractor;
-    // extractor.setLogType(ProgressLogger::NONE);
+    PeptideRefRTMap ref_rt_map;
+    for (auto& chunk : chunks)
     {
-      vector<OpenSwath::ChromatogramPtr> chrom_temp;
-      vector<ChromatogramExtractor::ExtractionCoordinates> coords;
-      extractor.prepare_coordinates(chrom_temp, coords, library_,
-          numeric_limits<double>::quiet_NaN(), false);
 
-      boost::shared_ptr<PeakMap> shared = boost::make_shared<PeakMap>(ms_data_);
-      OpenSwath::SpectrumAccessPtr spec_temp =
-        SimpleOpenMSSpectraFactory::getSpectrumAccessOpenMSPtr(shared);
-      extractor.extractChromatograms(spec_temp, chrom_temp, coords, mz_window_,
-          mz_window_ppm_, "tophat");
-      extractor.return_chromatogram(chrom_temp, coords, library_, (*shared)[0],
-          chrom_data_.getChromatograms(), false);
-    }
+      createAssayLibrary_(chunk.first, chunk.second, ref_rt_map);
+
+      /*if (debug_level_ >= 666)
```

Comment:
is it possible to move that out of the loop and do an 
extra createAssayLibrary( with begin and end? or is ref_rt_map changed?

Reply:
But that defeats the purpose of chunking, no?

Reply:
yes, we add to ref_rt_map in every iteration

Reply:
yes but this would only be intended for writing the information of the traml file

Reply:
so you could keep that debug traml thingy outside the loop